### PR TITLE
CAF-3646: Support list of prereq jobs

### PR DESCRIPTION
**Pull request by CJComac**

### CODE REVIEW

**Review by glucy — APPROVED** [review]

File: job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java [hunk 1]
```diff
@@ -158,6 +163,100 @@ private void testJobCompletion(final boolean useTaskDataObject) throws Exception
         }
     }
 
+    @Test
```
Comment (mulhollandc) [inline job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java:166]: Should we include an additional test here where all of the pre-requisite job Ids have completed to make sure no rows have made it into the job_task_data and job_dependency tables?

File: job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java [hunk 2]
```diff
@@ -300,6 +346,59 @@ public boolean doesJobAlreadyExist(String jobId, int jobHash) throws Exception {
     }
 
     /**
+     * Returns TRUE if the specified job id is complete, otherwise FALSE.
+     */
+    public boolean isJobComplete(final String jobId) throws Exception
```
Comment (mulhollandc) [inline job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java]: Why is isJobComplete() needed?

**Review by mulhollandc — CHANGES_REQUESTED** [review]

File: job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java (hunk 1 above)
Comment (CJComac) [reply job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java:166]: I have added this test in the latest commit

File: job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java (hunk 2 above)
Comment (CJComac) [reply job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java]: No longer needed, I have removed this from in latest commit

**Review by mulhollandc — APPROVED** [review]

**Pull request merged**